packages/node_modules/@webex/webex-core/src/lib/services/services.js

```diff
@@ -0,0 +1,374 @@
+   */
+  updateCredentialsConfig() {
+    const {idbroker, identity} = this.list(true);
+
+    if (idbroker && identity) {
+      this.webex.setConfig({
+        credentials: {
+          idbroker: {
+            url: idbroker.slice(0, -1) // remove trailing slash
+          },
+          identity: {
+            url: identity.slice(0, -1) // remove trailing slash
+          }
+        }
+      });
+    }
+  },
+
+  /**
+   * Wait until the service catalog is available,
+   * or reject afte ra timeout of 60 seconds.
+   * @param {string} serviceGroup
+   * @param {number} [timeout] - in seconds
+   * @returns {Promise<void>}
+   */
+  waitForCatalog(serviceGroup, timeout) {
+    const catalog = this._getCatalog();
+
+    if (serviceGroup === 'postauth' &&
+      this.webex.credentials.canAuthorize &&
+      !catalog.status.postauth.collecting
+    ) {
+      return this.updateServices();
+    }
+
+    return catalog.waitForCatalog(serviceGroup, timeout);
+  },
+
+  /**
+   * @private
+   * Organize a received hostmap from a service
+   * catalog endpoint.
+   * @param {object} serviceHostmap
+   * @returns {object}
+   */
+  _formatReceivedHostmap(serviceHostmap) {
+    const output = [];
+    let item = {};
+
+    Object.keys(serviceHostmap.serviceLinks).forEach((key) => {
+      item = {};
+
+      item.name = key;
+      item.defaultUrl = serviceHostmap.serviceLinks[key];
+      item.hosts = serviceHostmap.hostCatalog[Url.parse(item.defaultUrl).host];
+
+      output.push(item);
+    });
+
+    return output;
+  },
+
+  /**
+   * @private
+   * Simplified method wrapper for sending a request to get
+   * an updated service hostmap.
+   * @param {object} [param]
+   * @param {string} [param.from] - This accepts `limited` or `signin`
+   * @param {object} [param.query] - This accepts `email`, `orgId` or `userId` key values
+   * @param {string} [param.query.email] - must be a standard-format email
+   * @param {string} [param.query.orgId] - must be an organization id
+   * @param {string} [param.query.userId] - must be a user id
+   * @param {string} [param.token] - used for signin catalog
+   * @returns {Promise<object>}
+   */
+  _fetchNewServiceHostmap({from, query, token} = {}) {
+    const service = 'u2c';
+    const resource = from ? `/${from}/catalog` : '/catalog';
+    const qs = {...query, format: 'hostmap'};
+
+    const requestObject = {
+      method: 'GET', service, resource, qs
+    };
+
+    if (token) {
+      requestObject.headers = {authorization: token};
+    }
+
+    return this.request(requestObject)
+      .then(({body}) => this._formatReceivedHostmap(body));
+  },
+
+  /**
+   * Initializer
+   *
+   * @instance
+   * @memberof Services
+   * @returns {Services}
+   */
+  initialize() {
+    this._catalogs.set(this.webex, new ServiceCatalog());
+
+    this.listenToOnce(this.webex, 'ready', () => {
+      /* eslint-disable camelcase */
+      if (this.webex.credentials.canAuthorize) {
+        this.updateServices();
```

Comment:
Do we want to eventually call `updateServices` once `canAuthorize` is true? If so, should we do another `listenToOnce` in the else block?

Reply:
Yes, but, when performing other authorization flows outside of sending a token in with initialization, the correct time to call `updateServices()` changes.